app/controllers/api/automate_domains_controller.rb

```diff
@@ -67,17 +67,8 @@ def automate_domain_ident(domain)
       "Automate Domain id:#{domain.id} name:'#{domain.name}'"
     end
 
-    def resource_search(id, type, klass)
-      if id.to_s =~ /\A\d+\z/
-        super
-      else
-        begin
-          domain = collection_class(:automate_domains).find_by!(:name => id)
-        rescue
-          raise NotFoundError, "Couldn't find #{klass} with 'name'=#{id}"
-        end
-        super(domain.id, type, klass)
-      end
+    def resource_search(id, type, klass = nil, key_id = nil)
+      super(id, type, klass, id.to_s =~ /\A\d+\z/ ? :id : :name)
```

Comment:
```suggestion
      super(id, type, klass, id.to_s.integer? ? :id : :name)
```

Reply:
i see the old code was that way though, so no need to change for this PR.

Reply:
done. thanks